opentelemetry-sdk/src/trace/links.rs

```diff
@@ -0,0 +1,11 @@
+//! # Span Links
+
+use opentelemetry::trace::Link;
+/// Stores span links along with dropped count.
+#[derive(Clone, Debug, Default, PartialEq)]
```

Comment:
Better to annotate the struct with `#[non_exhaustive]` so that we can add more fields without breaking existing code

Reply:
If we don't compare them, I'd leave off the derive of `PartialEq`.

Reply:
Would it be appropriate to `impl Deref<Vec<Link>>` to avoid the double `span.span_links.links.` usage everywhere?

Reply:
Since SpanLinks is public I would assume we should keep it `PartialEq` because if none uses it then there's no impact, but if we don't include it then people would have to write a wrapper or extension any time they wanted to

Reply:
If we're implementing `Deref` (but not `DerefMut`) might make sense to `Deref` to `Target = &[Link]` instead of `Vec<Link>`.